bolt04/trampoline-payment-onion-test.json

```diff
@@ -0,0 +1,117 @@
+{
+    "comment": "test vector for a trampoline payment from Alice to Eve using Carol as intermediate trampoline node",
+    "generate_alice": {
+        "comment": "Alice creates a trampoline onion Carol -> Eve, and reaches Carol via Bob.",
+        "associated_data": "4242424242424242424242424242424242424242424242424242424242424242",
+        "trampoline_hops": [
+            {
+                "alias": "Carol",
+                "node_id": "027f31ebc5462c1fdce1b737ecff52d37d75dea43ce11c74d25aa297165faa2007",
+                "payload": {
+                    "amt_to_forward": 100000000,
+                    "outgoing_cltv": 800000,
+                    "outgoing_node_id": "02edabbd16b41c8371b92ef2f04c1185b4f03b6dcd52ba9b78d9d7c89c8f221145",
+                    "hex": "2e 020405f5e100 04030c3500 0e2102edabbd16b41c8371b92ef2f04c1185b4f03b6dcd52ba9b78d9d7c89c8f221145"
+                }
+            },
+            {
+                "alias": "Eve",
+                "node_id": "02edabbd16b41c8371b92ef2f04c1185b4f03b6dcd52ba9b78d9d7c89c8f221145",
+                "payload": {
+                    "amt_to_forward": 100000000,
+                    "outgoing_cltv": 800000,
+                    "payment_secret": "2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a",
+                    "total_amount_msat": 100000000,
+                    "hex": "31 020405f5e100 04030c3500 08242a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a2a05f5e100"
+                }
+            }
+        ],
+        "trampoline_session_key": "0303030303030303030303030303030303030303030303030303030303030303",
+        "trampoline_onion": "0002531fe6068134503d2723133227c867ac8fa6c83c537e9a44c3c5bdbdcb1fe3371860c0749bfd613056cfc5718beecc25a2f255fc7abbea3cd75ff820e9d30807d19b30f33626452fa54bb2d822e918558ed3e6714deb3f9a2a10895e7553c6f088c9a852043530dbc9abcc486030894364b205f5de60171b451ff462664ebce23b672579bf2a444ebfe0a81875c26d2fa16d426795b9b02ccbc4bdf909c583f0c2ebe9136510645917153ecb05181ca0c1b207824578ee841804a148f4c3df7306dcea52d94222907c9187bc31c0880fc084f0d88716e195c0abe7672d15217623",
+        "hops": [
+            {
+                "alias": "Bob",
+                "node_id": "0324653eac434488002cc06bbfb7f10fe18991e35f9fe4302dbea6d2353dc0ab1c",
+                "payload": {
+                    "amt_to_forward": 100005000,
+                    "outgoing_cltv": 800250,
+                    "short_channel_id": "0x0x1105",
```

Comment:
Note that I'm using the "standard human readable format" for scids here, defined here: https://github.com/lightning/bolts/blob/master/07-routing-gossip.md#definition-of-short_channel_id

But I'm not entirely sure that this is people are using in practice and comfortable with? Let me know if that's unclear, in which case I can either use the numeric value or its hex encoding (tell me what you think is best).